File: frontend-service/src/types/analytics.ts
// 分析服务响应格式
export interface AnalyticsResponse {
  success: boolean
  message: string
  data?: any
}

// 账户实时统计数据
export interface AccountRealtimeStats {
  active_accounts: number
  login_success_rate: number
  avg_response_time: number
  last_updated: string
  error?: string
}

// 账户历史统计数据
export interface AccountHistoryStats {
  time_range: string
  trends: AccountTrendData[]
  health_metrics: AccountHealthMetrics
  error?: string
}

export interface AccountTrendData {
  timestamp: string
  total_logins: number
  successful_logins: number
  success_rate: number
  avg_response_time: number
}

export interface AccountHealthMetrics {
  total_accounts: number
  active_accounts: number
  error_accounts: number
  health_score: number
}

// 代理实时统计数据
export interface ProxyRealtimeStats {
  total_proxies: number
  active_proxies: number
  success_rate: number
  avg_speed: number
  last_updated: string
  error?: string
}

// 代理历史统计数据
export interface ProxyHistoryStats {
  time_range: string
  trends: ProxyTrendData[]
  quality_metrics: ProxyQualityMetrics
  error?: string
}

export interface ProxyTrendData {
  timestamp: string
  total_requests: number
  successful_requests: number
  success_rate: number
  avg_speed: number
}

export interface ProxyQualityMetrics {
  total_proxies: number
  high_quality: number
  medium_quality: number
  low_quality: number
  avg_quality_score: number
}

// 任务实时统计数据
export interface TaskRealtimeStats {
  total_tasks: number
  running_tasks: number
  completed_tasks: number
  failed_tasks: number
  success_rate: number
  last_updated: string
  error?: string
}

// 任务历史统计数据
export interface TaskHistoryStats {
  time_range: string
  trends: TaskTrendData[]
  efficiency_metrics: TaskEfficiencyMetrics
  error?: string
}

export interface TaskTrendData {
  timestamp: string
  total_tasks: number
  completed_tasks: number
  failed_tasks: number
  success_rate: number
  avg_execution_time: number
}

export interface TaskEfficiencyMetrics {
  total_tasks: number
  avg_success_rate: number
  avg_execution_time: number
  resource_utilization: number
}

// 结果实时统计数据
export interface ResultRealtimeStats {
  total_results: number
  today_results: number
  week_results: number
  month_results: number
  last_updated: string
  error?: string
}

// 结果历史统计数据
export interface ResultHistoryStats {
  time_range: string
  trends: ResultTrendData[]
  quality_metrics: DataQualityMetrics
  error?: string
}

export interface ResultTrendData {
  timestamp: string
  total_results: number
  valid_results: number
  invalid_results: number
  quality_score: number
}

export interface DataQualityMetrics {
  total_results: number
  valid_results: number
  completeness_score: number
  accuracy_score: number
  consistency_score: number
}

// 代理质量统计
export interface ProxyQualityStats {
  proxy_id?: string
  success_rate: number
  avg_speed: number
  reliability_score: number
  usage_count: number
  last_used: string
  error?: string
}

// 任务效率统计
export interface TaskEfficiencyStats {
  task_id?: string
  success_rate: number
  avg_execution_time: number
  resource_usage: number
  completion_rate: number
  error?: string
}

// 数据质量洞察
export interface DataQualityInsights {
  collection_name?: string
  total_documents: number
  valid_documents: number
  completeness: number
  accuracy: number
  timeliness: number
  error?: string
}

// 账户使用统计
export interface AccountUsageStats {
  account_id?: string
  total_logins: number
  successful_logins: number
  avg_response_time: number
  last_login: string
  error?: string
}

// 活动类型枚举
export enum ActivityType {
  ACCOUNT_LOGIN = "ACCOUNT_LOGIN",
  PROXY_USAGE = "PROXY_USAGE",
  TASK_EXECUTION = "TASK_EXECUTION"
}

// 最近活动
export interface RecentActivity {
  id: string
  type: ActivityType
  timestamp: string
  description: string
  status: string
  details: Record<string, any>
  service: string
}

// 最近活动响应
export interface RecentActivitiesResponse {
  activities: RecentActivity[]
  total_count: number
}

// 任务历史统计数据响应格式
export interface TaskHistoryResponse {
  time_range: string
  trends: TaskTrendDataPoint[]
  total_count: number
  success_rate: number
  avg_execution_time: number
}

export interface TaskTrendDataPoint {
  timestamp: string
  total_tasks: number
  completed_tasks: number
  failed_tasks: number
  success_rate: number
  avg_execution_time: number
}

File: account-service/app/schemas/analytics.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime
from enum import Enum


class ActivityType(str, Enum):
    """活动类型枚举"""
    ACCOUNT_LOGIN = "ACCOUNT_LOGIN"
    PROXY_USAGE = "PROXY_USAGE"
    TASK_EXECUTION = "TASK_EXECUTION"


class AnalyticsResponse(BaseModel):
    """分析响应基础模型"""
    success: bool = Field(description="请求是否成功")
    message: str = Field(description="响应消息")
    data: Optional[Any] = Field(None, description="响应数据")


class RealtimeAnalyticsData(BaseModel):
    """实时分析数据"""
    active_accounts: int = Field(description="活跃账户数量")
    login_success_rate: float = Field(description="登录成功率 (0-1)")
    avg_response_time: float = Field(description="平均响应时间 (秒)")
    last_updated: datetime = Field(description="数据最后更新时间")


class HistoryAnalyticsData(BaseModel):
    """历史分析数据"""
    time_range: str = Field(description="时间范围")
    trends: List[Dict[str, Any]] = Field(description="趋势数据")
    health_metrics: Dict[str, Any] = Field(description="健康度指标")


class AccountUsageStats(BaseModel):
    """账户使用统计"""
    account_id: Optional[str] = Field(None, description="账户ID")
    username: Optional[str] = Field(None, description="用户名")
    total_logins: int = Field(description="总登录次数")
    successful_logins: int = Field(description="成功登录次数")
    failed_logins: int = Field(description="失败登录次数")
    success_rate: float = Field(description="成功率 (0-1)")
    avg_response_time: float = Field(description="平均响应时间 (秒)")
    last_login_time: Optional[datetime] = Field(None, description="最后登录时间")
    days_since_last_login: Optional[int] = Field(None, description="距离最后登录天数")


class AccountSummaryStats(BaseModel):
    """账户汇总统计"""
    total_accounts: int = Field(description="总账户数")
    active_accounts: int = Field(description="活跃账户数")
    inactive_accounts: int = Field(description="非活跃账户数")
    accounts_with_errors: int = Field(description="有错误的账户数")
    average_success_rate: float = Field(description="平均成功率 (0-1)")
    average_response_time: float = Field(description="平均响应时间 (秒)")
    inactive_days_threshold: int = Field(description="非活跃天数阈值")


class RecentActivity(BaseModel):
    """最近活动"""
    id: str = Field(description="活动ID")
    type: ActivityType = Field(description="活动类型")
    timestamp: datetime = Field(description="活动时间戳")
    description: str = Field(description="活动描述")
    status: str = Field(description="活动状态")
    details: Dict[str, Any] = Field(description="活动详情")
    service: str = Field(description="相关服务")


class HealthMetrics(BaseModel):
    """健康度指标"""
    total_accounts: int = Field(description="总账户数")
    active_accounts: int = Field(description="活跃账户数")
    error_accounts: int = Field(description="错误账户数")
    health_score: float = Field(description="健康度分数 (0-100)")


class LoginTrendData(BaseModel):
    """登录趋势数据"""
    time_bucket: datetime = Field(description="时间分桶")
    total_logins: int = Field(description="总登录次数")
    successful_logins: int = Field(description="成功登录次数")
    failed_logins: int = Field(description="失败登录次数")
    success_rate: float = Field(description="成功率 (0-1)")
    avg_response_time: float = Field(description="平均响应时间 (秒)")